File: domain/bot_status.go
package domain

type BotStatus int

const (
	Idle       BotStatus = 0
	Processing BotStatus = 1
)

func (bs BotStatus) String() string {
	switch bs {
	case Idle:
		return "Idle"
	case Processing:
		return "Processing"
	default:
		return "Unknown"
	}
}

func (bs BotStatus) IsIdle() bool {
	return bs == Idle
}

func (bs BotStatus) IsProcessing() bool {
	return bs == Processing
}

File: src/domain/bot/bot-status.ts
/**
 * A cooking bot is either waiting for work (IDLE) or busy cooking an order
 * (PROCESSING). A bot processes exactly one order at a time.
 */
export const BotStatus = {
  IDLE: 'IDLE',
  PROCESSING: 'PROCESSING',
} as const;

export type BotStatus = (typeof BotStatus)[keyof typeof BotStatus];
